### PR TITLE
[hotfix][test][formats] Detach TableCsvFormatITCase from JsonPlanTestBase

### PR DESCRIPTION
**Pull request by afedulov**
JsonPlanTestBase is not intended to be used for testing connectors and formats. This PR removes this dependency.

### CODE REVIEW

File: flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java [hunk 1]
```diff
@@ -18,32 +18,78 @@
...
+                        .column("a", BIGINT())
+                        .column("b", INT())
+                        .column("c", STRING())
+                        .build();
+
+        createSourceTable("MyTable", data, sourceSchema);
+
+        Schema sinkSchema = Schema.newBuilder().column("a", BIGINT()).column("c", STRING()).build();
+
+        File sinkPath = createSinkTable("MySink", sinkSchema);
```
Comment (alpreu) [inline flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java:92]: Can we just reuse sourceSchema here?

File: flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java [hunk 2]
```diff
@@ -165,30 +269,87 @@ private static String formatSqlTimestamp(long timestamp) {
...
+    private void assertResult(List<String> expected, File resultFile) throws IOException {
+        List<String> actual = readLines(resultFile);
+        assertResult(expected, actual);
+    }
+
+    private void assertResult(List<String> expected, List<String> actual) {
+        Collections.sort(expected);
+        Collections.sort(actual);
+        assertThat(actual).isEqualTo(expected);
+    }
```
Comment (alpreu) [inline flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java:338]: Using assertJ could be reduced to `assertThat(actual).hasSameElementsAs(expected);`

Comment (afedulov) [reply flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java:338]: Good catch, that was a residue from the JsonPlanTestBase.

File: flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java (hunk 1 above)
Comment (afedulov) [reply flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java:92]: This test checks that projection pushdown from the SQL engine works. Notice that after dropping `b` column from the sink schema, the returned elements are expected to also skip it.

**Comment by afedulov** [comment]
@alpreu thanks for the review!
I addressed your comments.

File: flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java (hunk 1 above)
Comment (alpreu) [reply flink-formats/flink-csv/src/test/java/org/apache/flink/formats/csv/TableCsvFormatITCase.java:92]: You're right, I overlooked the `b` part, sorry

**Review by alpreu — APPROVED** [review]

**Comment by afedulov** [comment]
@flinkbot run azure

**Pull request merged**